### PR TITLE
layers: Breakup Stateless validation

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Takes the 10,000 line `parameter_validation_utils.cpp` and breaks up into files to better represent the validation

### CODE REVIEW

File: layers/stateless/image_buffer_validation.cpp [hunk 1]
```diff
@@ -0,0 +1,794 @@
+/* Copyright (c) 2015-2023 The Khronos Group Inc.
```
Comment (arno-lunarg) [inline layers/stateless/image_buffer_validation.cpp:1]: Would it make sense to have both `stateless/image_validation.cpp` and `stateless/buffer_validation.cpp`?

File: layers/stateless/instance_device_validation.cpp [hunk 2]
```diff
@@ -0,0 +1,866 @@
+/* Copyright (c) 2015-2023 The Khronos Group Inc.
```
Comment (arno-lunarg) [inline layers/stateless/instance_device_validation.cpp:1]: Same question here with `stateless/instance_validation.cpp` and `stateless_device_validation.cpp`

File: layers/stateless/render_pass_validation.cpp [hunk 3]
```diff
@@ -0,0 +1,521 @@
...
+}
+
+void StatelessValidation::PostCallRecordDestroyRenderPass(VkDevice device, VkRenderPass renderPass,
+                                                          const VkAllocationCallbacks *pAllocator) {
+    // Track the state necessary for checking vkCreateGraphicsPipeline (subpass usage of depth and color attachments)
+    std::unique_lock<std::mutex> lock(renderpass_map_mutex);
+    renderpasses_states.erase(renderPass);
+}
+
+bool StatelessValidation::manual_PreCallValidateCreateFramebuffer(VkDevice device, const VkFramebufferCreateInfo *pCreateInfo,
```
Comment (arno-lunarg) [inline layers/stateless/render_pass_validation.cpp:511]: What about a dedicated `stateless/framebuffer_validation.cpp`?

File: layers/stateless/image_buffer_validation.cpp (hunk 1 above)
Comment (spencer-lunarg) [reply layers/stateless/image_buffer_validation.cpp:1]: I guess I saw the `buffer_validation` as 2 functions and only like 80 lines of code... but honestly that is how we got into these 10k line files in the first place... will split

File: layers/stateless/instance_device_validation.cpp (hunk 2 above)
Comment (spencer-lunarg) [reply layers/stateless/instance_device_validation.cpp:1]: this one is harder because some of these helper functions start to span both

**Review by ncesario-lunarg — APPROVED** [review]

**Pull request merged**